src/script/miniscript.h

```diff
@@ -940,6 +962,33 @@ struct Node {
                     Availability avail = ctx.Sign(node.keys[0], sig);
                     return {ZERO + InputStack(key), (InputStack(std::move(sig)).SetWithSig() + InputStack(key)).SetAvailable(avail)};
                 }
+                case Fragment::MULTI_A: {
+                    // sats[j] represents the best stack containing j valid signatures (out of the first i keys).
+                    // In the loop below, these stacks are built up using a dynamic programming approach.
+                    std::vector<InputStack> sats = Vector(EMPTY);
+                    // The dissatisfaction consists of as many empty vectors as there are keys.
+                    InputStack nsat;
+                    for (size_t i = 0; i < node.keys.size(); ++i) {
+                        nsat = std::move(nsat) + ZERO;
+                        // Get the signature for the i'th key in reverse order (the signature for the first key needs to
+                        // be at the top of the stack, contrary to CHECKMULTISIG's satisfaction).
+                        std::vector<unsigned char> sig;
+                        Availability avail = ctx.Sign(node.keys[node.keys.size() - 1 - i], sig);
+                        // Compute signature stack for just this key.
+                        auto sat = InputStack(std::move(sig)).SetWithSig().SetAvailable(avail);
+                        // Compute the next sats vector: next_sats[0] is a copy of sats[0] (no signatures). All further
+                        // next_sats[j] are equal to either the existing sats[j], or sats[j-1] plus a signature for the
+                        // current (i'th) key. The very last element needs all signatures filled.
+                        std::vector<InputStack> next_sats;
+                        next_sats.push_back(sats[0] + ZERO);
+                        for (size_t j = 1; j < sats.size(); ++j) next_sats.push_back((sats[j] + ZERO) | (std::move(sats[j - 1]) + sat));
+                        next_sats.push_back(std::move(sats[sats.size() - 1]) + std::move(sat));
+                        // Switch over.
+                        sats = std::move(next_sats);
+                    }
+                    assert(node.k <= sats.size());
+                    return {std::move(nsat), std::move(sats[node.k])};
```

Comment:
For my code you are right, i don't need `nsat`, i can juste reuse `sats[0]`. I made the change, it's straightforward and i think cleaner that way.

For `multi` however we can't do this since `sats[0]` doesn't have `ZERO`s, since `CHECKMULTISIG` does not need a `ZERO` as signatures for keys that are unused in case of satisfaction (it will just try the next public key for the same signature in case of failure). I haven't looked into `thresh`, but i suppose we could (i mostly reused `thresh`'s algorithm for `multi_a`).